app/scripts/model/dataProxy.js

```diff
@@ -340,28 +357,39 @@
             _osSurvivalAttrMeta.display_name = 'Overall Survival';
             _osSurvivalAttrMeta.filter = [];
             _osSurvivalAttrMeta.show = true;
-            _patientAttributes.unshift(_osSurvivalAttrMeta);
+            _osSurvivalAttrMeta.priority = 1;
+            _patientAttributes[_osSurvivalAttrMeta.attr_id] = _osSurvivalAttrMeta;
           }
 
           // add Cancer Study
           if (self.getCancerStudyIds().length > 1) {
-            _patientAttributes.unshift({
+            _patientAttributes['study_id'] = {
```

Comment:
Should we add keys, numOfDatum, priority attributes to this meta object and even missing keys, numOfDatum to the above charts(tables survival and scatter meta).